### PR TITLE
[Pygen]: Fix minor issues

### PR DESCRIPTION
**Pull request by aneels3**
Add fix for ```randcase``` TODO.
Fixed other minor issues related to RV32B and illegal_instr_test.

### CODE REVIEW

**Comment by shrujal20** [comment]
@googlebot I consent.

File: pygen/pygen_src/riscv_asm_program_gen.py [hunk 1]
```diff
@@ -437,15 +437,35 @@ def init_floating_point_gpr_with_dpf(self, int_floating_gpr):
         self.instr_stream.extend((li_instr0, slli_instr, li_instr1, or_instr, fmv_instr))
 
     # Get a random single precision floating value
+    @staticmethod
+    def get_randselect0(addr_range1, addr_range2):
```
Comment (hodjat91) [inline pygen/pygen_src/riscv_asm_program_gen.py:441]: why is it get_randselect"0"? Can we use a better name?

File: pygen/pygen_src/riscv_asm_program_gen.py [hunk 2]
```diff
@@ -437,15 +437,35 @@ def init_floating_point_gpr_with_dpf(self, int_floating_gpr):
...
+        value = vsc.rand_bit_t(32)
+        with vsc.randomize_with(value): value in vsc.rangelist(addr_range1, addr_range2)
+
     def get_rand_spf_value(self):
-        # TODO randcase
-        value = random.randrange(0, 2**32 - 1)
+        vsc.randselect([
+            (1, lambda: get_randselect0(hex(0x7f80_0000), hex(0xff80_0000))),
```
Comment (hodjat91) [inline pygen/pygen_src/riscv_asm_program_gen.py:447]: Maybe adding a comment here would be great (about the addr_range1 and addr_range2)

File: pygen/pygen_src/riscv_asm_program_gen.py (hunk 1 above)
Comment (hodjat91) [reply pygen/pygen_src/riscv_asm_program_gen.py:441]: I see there is another version of this function for 64 bits. Can you please combine the two methods?

File: pygen/pygen_src/riscv_asm_program_gen.py [hunk 3]
```diff
@@ -437,15 +437,35 @@ def init_floating_point_gpr_with_dpf(self, int_floating_gpr):
...
-    def get_rand_dpf_value(self):
+    @staticmethod
+    def get_randselect1(addr_range1, addr_range2):
         value = vsc.bit_t(64)
-        # TODO randcase
+        with vsc.randomize_with(value): value in vsc.rangelist(addr_range1, addr_range2)
+
+    def get_rand_dpf_value(self):
+        vsc.randselect([
+            (1, lambda: get_randselect1(hex(0x7ff0_0000_0000_0000), hex(0xfff0_0000_0000_0000))),
```
Comment (udinator) [inline pygen/pygen_src/riscv_asm_program_gen.py:463]: same here (re: adding comments about the address ranges), linking to the appropriate area of the spec would work as well IMO

**Review by udinator — APPROVED** [review]

**Pull request merged**